ocaml/typing/outcometree.mli

```diff
@@ -111,7 +101,31 @@ type out_ret_mode =
   (** The ret type is arrow, and need to print parens around the arrow, with
       modes annotating. *)
 
-type out_type =
+(** Represents a user-written jkind annotation *)
+type out_jkind_user =
+  | Ojkind_user_default
+  | Ojkind_user_abbreviation of string
+  | Ojkind_user_mod of out_jkind_user * string list
+  | Ojkind_user_with of out_jkind_user * out_type
+  | Ojkind_user_kind_of of out_type
+
+and out_jkind_const = { base : string; modal_bounds : string list }
+
+and out_jkind =
+  | Ojkind_user of out_jkind_user
+  | Ojkind_const of out_jkind_const
+  | Ojkind_var of string
```

Comment:
This doesn't quite seem right: `Ojkind_user (Ojkind_user_abbreviation ...)` is the same as `Ojkind_var ...`. I think we shouldn't need to care about whether the user wrote the annotation or not. Let's talk if there is trouble simplifying this.

Reply:
Fixed